r2/r2/controllers/api.py

```diff
@@ -883,7 +887,16 @@ def POST_unfriend(self, nuser, iuser, container, type):
                 contributor='removecontributor',
                 muted='unmuteuser',
             ).get(type, None)
-            ModAction.create(container, c.user, action, target=victim)
+            details = None
+            if (c.user == victim and
```

Comment:
This `if` is extremely difficult to follow (and `is` shouldn't be used to check string equality).

I really don't think it's even necessary, I'd probably just do `if c.user == victim and c.site.is_moderator(c.user):` and set the `remove_self` note on that. I can't see any way that the distinction of "moderator that has permissions to remove others is removing themselves" vs. "user is a moderator that doesn't have permissions to remove others, but is leaving this status" is important at all.